### ISSUE TITLE
IBM test suite build failures on v4.0

### ISSUE THREAD

**Issue by bwbarrett** [open-mpi/ompi#6149]
MTT is giving build fails on all IBM test suite runs on the IBM test suite.  An example from AWS's results shows the output below.  Tagging as a blocker, as I haven't verified whether this is a regression against 4.0 or not.

```
CCLD     libattrsetget.la
  FCLD     comm_idup_usempi
  FCLD     create_group_usempi
  CCLD     self_atexit
attr_mpi1.o: In function `main':
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:58: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:65: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:71: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:78: undefined
reference to `OMPI_C_MPI_NULL_DELETE_FN'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:78: undefined
reference to `OMPI_C_MPI_NULL_COPY_FN'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:78: undefined
reference to `MPI_Keyval_create'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:85: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:94: undefined
reference to `MPI_Attr_put'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:96: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:105: undefined
reference to `MPI_Keyval_free'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:112: undefined
reference to `MPI_Attr_get'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:126: undefined
reference to `MPI_Attr_delete'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:130: undefined
reference to `MPI_Attr_delete'
/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator/attr_mpi1.c:134: undefined
reference to `MPI_Attr_delete'
collect2: error: ld returned 1 exit statusmake[1]: *** [attr_mpi1] Error 1
make[1]: *** Waiting for unfinished jobs....make[1]: Leaving directory
`/home/ec2-user/mtt-scratch/installs/s_VB/tests/ibm/ibm/communicator'
make: *** [all-recursive] Error 1
```

**Comment by bosilca** [comment]
These are all the MPI1 deprecated symbols. I guess we need to update the IBM tests to check for the existence of such deprecated symbols before trying to use them.

**Comment by bwbarrett** [comment]
@bosilca yeah, we figured that part out, but @jsquyres had thought we had done all the right update bits.

**Comment by bwbarrett** [comment]
Digging into the failure, this is an artifact of https://github.com/open-mpi/ompi/pull/6120 having not been completed.  This is definitely a blocker for 4.0.1.

The IBM test suite only checks that the MPI-1 deprecated / removed symbols are declared in mpi.h, with no check on compile.  They're weird tests, but they're catching the real issue, so maybe that's not awful.  Either way, once https://github.com/open-mpi/ompi/pull/6120 is completed, this bug can be closed.

**Comment by ggouaillardet** [comment]
I already have a patch for the test-suite I will cleanup and push.

FWIW, one issue is we test the existence of some subroutines in the library (symbols are still present in `libmpi.so`) without checking the symbol in the header file (prototypes are #ifdef'ed out from `mpi.h` unless Open MPI was configure'd with `--enable-mpi1-compatibility`)

**Comment by ggouaillardet** [comment]
I should have checked first ... I already pushed my changes in open-mpi/ompi-tests#89 and open-mpi/ompi-tests#90 and will double check everything on my side.

There is clearly a blocker in Open MPI v4 that prevents 4.0.1 from being released, and I was pointing to a different and non critical issue in the test suite.

Anyway, I am double checking everything now.

**Comment by ggouaillardet** [comment]
@bwbarrett I am sorry I mixed up two different issues.
 
In this case, some deprecated symbols are still in the header files but were removed from `libmpi.so` by mistake.  This is really an Open MPI v4 issue.

**Comment by gpaulsen** [comment]
Has this been addressed in v4.0.0rc1 ?

**Comment by hppritcha** [comment]
this should have been fixed.  Closing.  reopen if it's still a problem in v4.0.1
